include/openPMD/regions/Box.hpp

```diff
@@ -0,0 +1,437 @@
+#ifndef REGIONS_BOX_HPP
+#define REGIONS_BOX_HPP
+
+#include "Helpers.hpp"
+#include "Point.hpp"
+
+#include <array>
+#include <cstddef>
+#include <cstdlib>
+#include <functional>
+#include <iostream>
+#include <tuple>
+#include <vector>
+
+namespace openPMD {
+namespace Regions {
+
+/** A D-dimensional box
+ *
+ * A box is described by two points, its lower bound (inclusive) and
+ * upper bound (exclusive). If the lower bound not less than the upper
+ * bound, the box is empty. Empty boxes are fine (similar to an empty
+ * array).
+ *
+ * The dimension D needs to be known at compile time. @see NDBox
+ */
+template <typename T, std::size_t D> class Box;
+
+template <typename T> class Box<T, 0> {
+  bool is_full;
+
+  explicit constexpr Box(bool is_full_) : is_full(is_full_) {}
+
+public:
+  constexpr static std::size_t D = 0;
+
+  typedef typename Point<T, D>::value_type value_type;
```

Comment:
Do we have a standard on `typedef` vs. `using`? I think we mostly use `using` elsewhere, but I don't think that it is specified anywhere.

Reply:
Done.